spotty/spotty/UIComponents/Constraints/LayoutAnchor.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  LayoutAnchor.swift
+//  spotty
+//
+//  Created by Ivaylo Kalaydzhiev on 20.06.22.
+//
+
+import UIKit
+
+/// Represents a single `NSLayoutConstraint`
+enum LayoutAnchor {
+    
+    case constant(attribute: NSLayoutConstraint.Attribute,
+                  relation: NSLayoutConstraint.Relation,
+                  constant: CGFloat)
+
+    case relative(attribute: NSLayoutConstraint.Attribute,
+                  relation: NSLayoutConstraint.Relation,
+                  relatedTo: NSLayoutConstraint.Attribute,
+                  multiplier: CGFloat,
+                  constant: CGFloat)
+}
+
+// MARK: - Factory methods
+extension LayoutAnchor {
+    
+    static let leading = relative(attribute: .leading, relation: .equal, relatedTo: .leading)
+    static let trailing = relative(attribute: .trailing, relation: .equal, relatedTo: .trailing)
+    static let top = relative(attribute: .top, relation: .equal, relatedTo: .top)
+    static let bottom = relative(attribute: .bottom, relation: .equal, relatedTo: .bottom)
+
+    static let centerX = relative(attribute: .centerX, relation: .equal, relatedTo: .centerX)
+    static let centerY = relative(attribute: .centerY, relation: .equal, relatedTo: .centerY)
+
+    static let width = constant(attribute: .width, relation: .equal)
+    static let height = constant(attribute: .height, relation: .equal)
+
+    static func constant(
```

Comment:
Good idea. Enrich it with documentation.

Reply:
done